### PR TITLE
Added 3.0.0-beta-4 changelog.

### PR DESCRIPTION
**Pull request by xfxyjwf**
@pherl @haberman to review.

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -1,3 +1,83 @@
...
+  * Added google::protobuf::Map::swap() to swap two map fields.
+  * Fixed a memory leak when calling Reflection::ReleaseMessage() on a message
+    allocated on arena.
+  * Improved error reporting when parsing text format protos.
+  * JSON
+      - Added a new parser option to ignore unknown fields when parsing JSON.
+      - Added convenient methods for message to/from JSON conversion.
+  * Various performance optimizations.
+
+  Java (beta)
```
Comment (pherl) [inline CHANGES.txt]: What about lite runtime, we should mention that it's excluded from the main generator (are we doing the change in beta4?)

Comment (xfxyjwf) [inline CHANGES.txt]: Added a mentioning of Java lite. It's removed and will not be supported in this beta-4 release.

**Comment by xfxyjwf** [comment]
Ping. Also removed Java deterministic API from this PR.

**Comment by pherl** [comment]
LG

**Pull request merged**